agent/engine/docker_container_engine.go

```diff
@@ -519,6 +522,39 @@ func (dg *DockerGoClient) ContainerEvents(ctx context.Context) (<-chan DockerCon
 	return changedContainers, nil
 }
 
+// ListContainers lists returns a slice of container IDs.
+func (dg *DockerGoClient) ListContainers(all bool) ListContainersResponse {
+	timeout := ttime.After(listContainersTimeout)
+
+	response := make(chan ListContainersResponse, 1)
+	go func() { response <- dg.listContainers(all) }()
+	select {
+	case resp := <-response:
+		return resp
+	case <-timeout:
+		return ListContainersResponse{Error: &DockerTimeoutError{listContainersTimeout, "listing"}}
+	}
+}
+
+func (dg *DockerGoClient) listContainers(all bool) ListContainersResponse {
+	client := dg.dockerClient
+
+	// Empty options object will get us only running containers.
```

Comment:
I think this comment might be from before 'all' was an argument to this? Seems out of place now.